docker-entrypoint/dependencies/entrypoint.go

```diff
@@ -0,0 +1,80 @@
+package entrypoint
+
+import (
+	"github.com/stackanetes/docker-entrypoint/logger"
+	//	"k8s.io/kubernetes/pkg/client/restclient"
+	client "k8s.io/kubernetes/pkg/client/unversioned"
+	"os"
+	"sync"
+	"time"
+)
+
+var (
+	Dependencies     []Resolver      // List containing all dependencies to be resolved
+	DependencyPrefix = "DEPENDENCY_" //Prefix for env variables
+	Namespace        string          //Namespace for containers
+	wg               sync.WaitGroup
+	INTERVAL         = 2
+)
+
+// Object containing k8s client
+type Entrypoint struct {
+	Client *client.Client
+}
+
+//Constructor for entrypoint
+func NewEntrypoint() (entry Entrypoint) {
```

Comment:
I would probably write it more like this:

```
func NewEntrypoint(client *client.Client) (entry Entrypoint, err error) {
	if client == nil {
		if entry.Client, err = client.NewInCluster(); err != nil {
			return fmt.Errorf("Error while creating k8s client: %s", err)
		}
	}
	
	entry.Namespace = os.Getenv("NAMESPACE")
	if entry.Namespace == "" {
		logger.Warning.Print("NAMESPACE env not set, using default")
		entry.Namespace = "default"
	}
	
	return
}
```

It allows external users to specify a client if they don't want to use the default InCluster, it also defines a default namespace. And more importantly, it does not exit the app. If it is used as a library, users won't expect it.

As a side note, we should probably manipulate the object as a pointer everywhere (not done above).